### PR TITLE
New Expression should accept trailing commas Fixes #142

### LINKED ISSUE #142: Fails to parse trailing coma in constructor call
```
> new (require('cst').Parser)().parse('new Promise(executor,)')
Error: Expression expected but "Punctuator" found
    at ElementAssert._passExpressionInParens (/home/graingert/projects/procensus/frontend/node_modules/cst/src/elements/ElementAssert.js:316:19)
    at ElementAssert.passExpression (/home/graingert/projects/procensus/frontend/node_modules/cst/src/elements/ElementAssert.js:256:21)
    at NewExpression._acceptChildren (/home/graingert/projects/procensus/frontend/node_modules/cst/src/elements/types/NewExpression.js:28:40)
    at NewExpression._setChildren (/home/graingert/projects/procensus/frontend/node_modules/cst/src/elements/Element.js:639:14)
    at NewExpression.Element (/home/graingert/projects/procensus/frontend/node_modules/cst/src/elements/Element.js:50:18)
    at NewExpression.Node (/home/graingert/projects/procensus/frontend/node_modules/cst/src/elements/Node.js:11:33)
    at NewExpression.Expression (/home/graingert/projects/procensus/frontend/node_modules/cst/src/elements/Expression.js:13:33)
    at new NewExpression (/home/graingert/projects/procensus/frontend/node_modules/cst/src/elements/types/NewExpression.js:4:29)
    at buildElementTreeItem (/home/graingert/projects/procensus/frontend/node_modules/cst/src/elementTree.js:146:24)
    at buildElementTreeItem (/home/graingert/projects/procensus/frontend/node_modules/cst/src/elementTree.js:119:45)
```

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by graingert** [comment]
@markelog ready for review!

@hzoo you made the change originally for CallExpression so you're probably best placed to review this.

**Comment by graingert** [comment]
@hzoo @markelog https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb-base/CHANGELOG.md#1000--2016-11-06 has released now. and that requires code incompatible with cst.

**Comment by markelog** [comment]
So are you planning to finish this soon?

**Comment by graingert** [comment]
@markelog it's already done, as far as I'm aware.

**Comment by graingert** [comment]
@markelog nope

**Comment by graingert** [comment]
nocomments

File: src/elements/types/utils/BaseCallExpression.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import Expression from '../../Expression';
+
+export default class BaseCallExpression extends Expression {
```
Comment (graingert) [inline src/elements/types/utils/BaseCallExpression.js]: sure, do you want to?

**Comment by markelog** [comment]
Wow, terrible, it seems I wasn't adapted to new UI github changes.

Thank you!

File: test/lib/elements/types/NewExpression.js [hunk 2]
```diff
@@ -1,50 +1,31 @@
...
-        expect(expression.arguments.length).to.equal(0);
-    });
-
-    it('should accept single argument', () => {
-        var expression = parseAndGetExpression('new x ( x )');
-        expect(expression.callee.name).to.equal('x');
-        expect(expression.arguments.length).to.equal(1);
-        expect(expression.arguments[0].type).to.equal('Identifier');
-        expect(expression.arguments[0].name).to.equal('x');
+    it('should disallow super', () => {
```
Comment (graingert) [inline test/lib/elements/types/NewExpression.js]: fml that's handy

File: src/elements/types/utils/BaseCallExpression.js (hunk 1 above)
Comment (markelog) [inline src/elements/types/utils/BaseCallExpression.js]: Ooh, heh :). Yes, please

**Comment by graingert** [comment]
@markelog done!

**Comment by graingert** [comment]
@markelog any chance of a 0.4.7 release of this soon?

**Comment by markelog** [comment]
Done!

**Comment by graingert** [comment]
Shweet

**Pull request closed without merging**